auto-cpufreq.py

```diff
@@ -148,24 +145,47 @@ def set_powersave():
     print("Total system load:", load1m, "\n")
 
     # conditions for setting turbo in powersave
-    if load1m > cpus:
+    if load1m >= cpus * (3 / 5):
+        print("Setting to use: \"performance\" governor")
+        s.run("cpufreqctl --governor --set=performance", shell=True)
+
```

Comment:
I'm not sure about switching to "performance" while in "powersave" mode, regardless of the load/CPU usage. 

Only thing that would give us valuable metric would be to test how much battery would be used on battery while in "peformance" mode on battery vs  given same conditions in "powersave". I believe more battery will be used in "performance".

Hence the reason why I went with option to turn turbo boost on instead, which should handle the load.

Also, based on what criteria did you come up with this number: `load1m >= cpus * (3 / 5)`

Reply:
I just want to sure about the user's CPU load is more than 'more than the average'. Because if there is a very high load, even though the discharging, we should provide to the user the `performance` governor. The `3/5` is changeable though.

Reply:
After thinking about this for couple of days now, let's do this. Currently `auto-cpufreq` is really "trigger ready". In sense that it will easily switch governors and turn on/off turbo.

I like what you did here with calculations, but instead of switching to "performance" governor while on battery, let it turn on/off turbo instead depending on load/cpu usage while in powersave (battery discharging). 

This way we'll have refined way of turning turbo on/off and tool being less of "trigger friendly".

Reply:
@AdnanHodzic I made some changes about your explanation. Could you have a chance for looking at it?

Reply:
Right, but currently as part of `set_performance` function it's still set to change governors, which is not good.

```
        print("Setting to use: \"powersave\" governor")
        s.run("cpufreqctl --governor --set=powersave", shell=True)

        print("Load optimal, setting turbo boost: on")
        s.run("echo 0 > /sys/devices/system/cpu/intel_pstate/no_turbo", shell=True)
```

Which results in situations like these:

```
Battery is: charging

Total CPU usage: 2.0 %
Total system load: 0.24 

Setting to use: "powersave" governor
Load optimal, setting turbo boost: on
```

Why is turbo on if load is optimal? Ass I said, let's not change governors if it's charging (performance) or discharging (powersave).

Once in one of those 2 governors, let's just turn turbo on/off based on more refined cpu usage/system load.

Reply:
Well, I think if there is no or small load of CPU, pushing the clock to the best base clock freq is not necessary, which is performance governor does. If we are very keen on power saving when low load, I think that way is more accurate because powersave governor could lower the freq of the CPU and if necessary, it will push the clock freq to the higher.

Reply:
I'm sorry, but I don't agree. There is "small load", let's say you're on battery power and hence in powersave, you're watching 1 minute 4K Youtube video. Turbo kicks in, does the heavy load once the load is gone, it turns off turbo. Battery was preserved to its max, while CPU temperature/fans stayed at their optimal levels.

Only exception, where I agree with you that app should switch from powersave to performance (while on battery) is if we could figure out some kind of algorithm or equation where it detects it's reallynecessary. For example, if load and cpu usage (and in future CPU temp/fan speed) are running high for 5 minutes, besides turbo also switch the governor form powersave to performance.

Otherwise if you have app switching to performance while on battery battery for every little thng, battery life will suffer just as it would without this tool, which then defeats its purpose.

But, these are things that can be done at later stages. Right now, I think it would be the best if we could refine the current settings under which "turbo" is turned on/off and make the app less "trigger friendly", which you already did with i.e: ` if load1m > cpus / 7:`

But let's do it once step at a time is what I'm suggesting here.

Reply:
> For example, if load and cpu usage (and in future CPU temp/fan speed) are running high for >= 5 minutes, besides turbo also switch the governor from powersave to performance.

That looks easily doable.

Other than, I have already reverted what did you say of "performance governor when battery" situation. I think you understand my last comment wrong. I said these words for "powersave governor with turbo boost capability -it doesn't mean we push the clock to highest- with when charging". Could you have a chance for giving an explanation for your thoughts? Why performance with turbo boost capability off (that means we are pushing the clock to base speed which is the 1.6 GHz for my case) instead of powersave with turboboost capability enabled (no low load -> 0.8-1.2 GHz, if load it increases but it drops quickly when not necessary)?